NASA-AMMOS/aerie-ui
# Scheduling Subscription and Canceling

Subscribe to scheduling runs, refine computation and display of scheduling status, and allow users to cancel the current scheduling request if pending or incomplete.

Changes:
- Subscribe to the `scheduling_requests` table, filtered by current scheduling spec id
- Subscribe to current scheduling spec
- Refactor `effects.schedule` to use a temporary subscription on the `schedulingRequests` store (which is now a gql subscription) instead of polling the `schedule` action
- Display modified status for scheduling when out of date with scheduling spec or plan revision

Closes #1040, closes #1047.

Testing:
1. Make a plan
2. Verify that scheduling status in the navbar is not present as no scheduling runs have been performed
3. Add a scheduling goal to the plan in a new tab that should add activities 
4. Verify that the scheduling status for the plan is marked as modified without having to refresh the page
5. Run scheduling analysis
6. Scheduling status should transition to in-progress and then to failure
7. Inside the plan's scheduling panel the goal should indicate that not all goals were satisfied
8. Verify that the navbar status on hover indicates 1 unsatisfied goal
9. Run scheduling
10. Scheduling status should transition to in-progress and then to success
11. Refresh the page and ensure that the status remains success
12. Modify the plan (move an activity)
13. Re-run scheduling
14. Ensure that a new simulation is loaded in the UI (check simulation panel) after scheduling completes
15. Modify the scheduling goal in the other tab
16. Ensure that the scheduling status is now modified
